engine/tester/config_parse.c

```diff
@@ -996,6 +1000,144 @@ get_test_attrs(xmlNodePtr node, test_attrs *attrs)
     return 0;
 }
 
+/**
+ * Get default argument values for script.
+ *
+ * @param node        XML node with script call description
+ * @param session     Location for session description
+ *
+ * @return Status code.
+ */
+static te_errno
+get_script_defaults(xmlNodePtr node, const test_session *session)
+{
+    te_errno        rc;
+    script_def_arg *def_arg;
+    script_def_arg *darg;
+    char           *defaults_id;
+
+    def_arg = TE_ALLOC(sizeof(*def_arg));
+    TAILQ_INIT(&def_arg->args);
+
+    defaults_id = XML2CHAR(xmlGetProp(node, CONST_CHAR2XML("id")));
+    if (defaults_id == NULL)
+    {
+        ERROR("'id' attribute is missing in script with default argument "
+              "values");
+        return TE_RC(TE_TESTER, TE_EINVAL);
+    }
+
+    def_arg->id = defaults_id;
+    def_arg->name = XML2CHAR(xmlGetProp(node, CONST_CHAR2XML("name")));
+
+    node = xmlNodeChildren(node);
+
+    /* Get information about arguments */
+    while (node != NULL &&
+           xmlStrcmp(node->name, CONST_CHAR2XML("arg")) == 0)
+    {
+        rc = alloc_and_get_var_arg(node, false, session, &def_arg->args);
+        if (rc != 0)
+        {
+            ERROR("Processing default argument values of with script id '%s' "
+                  "failed: %r", defaults_id, rc);
+            return rc;
+        }
+        node = xmlNodeNext(node);
+    }
+
+    if (node != NULL)
+    {
+        ERROR("Unexpected element '%s' in default argument values for script "
+              "id '%s' call description", XML2CHAR(node->name), defaults_id);
+        return TE_RC(TE_TESTER, TE_EINVAL);
+    }
+
+    TAILQ_FOREACH(darg, &tester_global_context.def_args, links)
+    {
+        if (strcmp(darg->id, def_arg->id) == 0)
+        {
+            ERROR("Default value id '%s' is not unique.", def_arg->id);
+            return TE_RC(TE_TESTER, TE_EINVAL);
+        }
+    }
+    TAILQ_INSERT_TAIL(&tester_global_context.def_args, def_arg, links);
+    VERB("Got script defaults with id '%s'", defaults_id);
+
+    return 0;
+}
+
+/**
+ * Get default argument by default value ID.
+ *
+ * @param id        Default value ID
+ * @param def_arg   Location of default argument structure
+ *
+ * @return Status code.
+ */
+static te_errno
+get_default_arg_by_id(const char *id, script_def_arg **def_arg)
+{
+    script_def_arg *darg;
+
+    TAILQ_FOREACH(darg, &tester_global_context.def_args, links)
+    {
+        if (strcmp(darg->id, id) == 0)
+        {
+            *def_arg = darg;
+            return 0;
+        }
+    }
+    ERROR("There is no default argument value with ID '%s'", id);
+    return TE_RC(TE_TESTER, TE_ENOENT);
+}
+
+/**
+ * Add default argument values to run item argument list.
+ *
+ * @param id      Default value ID
+ * @param list    List of run item arguments
+ *
+ * @return Status code.
+ */
+static te_errno
+add_default_arg(const char *id, test_vars_args *list)
+{
+    te_errno        rc;
+    script_def_arg *def_arg = NULL;
+    bool            found;
```

Comment:
The declaration may be moved at line 1119, since the variable seems to be unused outside of the outer `TAILQ_FOREACH`.